kolibri/core/device/middleware.py

```diff
@@ -1,17 +1,70 @@
+from django.conf import settings
 from django.http import HttpResponse
-from django.middleware.locale import LocaleMiddleware
+from django.http import HttpResponseRedirect
+from django.urls import is_valid_path
 from django.utils import translation
+from rest_framework.views import APIView
 
 from .translation import get_language_from_request
+from kolibri.utils.conf import OPTIONS
 
 
-class KolibriLocaleMiddleware(LocaleMiddleware):
+class KolibriLocaleMiddleware:
```

Comment:
Why was the inheritance removed? It would be okay to override all the current methods, but keeping the inheritance somehow also means keeping the contract of what a `LocaleMiddleware` is supposed to do.

Reply:
@benjaoming I think the reasons are explained in the related PR #4984 :

> Also cleans up some unused code (including the sessions middleware which is a no op in Django - the presence of it in settings is only used to trigger some backwards compatibility behaviour that we do not need).

Reply:
I think it's better to keep it and override changed/unneeded methods. Simply because of maintaining the link to the reference implementation.

Reply:
The other reason for not inheriting is that the default Django locale middleware class is an old style middleware, so I would be overriding its methods simply to turn them into a no op (and I am not entirely sure how using new style middleware works when inheriting from the middleware deprecation compatibility class).